### PR TITLE
feat: Deprecate ActionSteps

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Needs https://github.com/PostHog/posthog/pull/22092

WARNING: Should only be merged once the dependent PRs are in and we are fully off of using ActionStep

## Changes

* Removes the use of the model everywhere

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/models/action/action.py [hunk 1]
```diff
@@ -72,24 +72,8 @@ def get_analytics_metadata(self):
 
     @property
     def steps(self) -> list[ActionStepJSON]:
-        if self.steps_json is None:
```
Comment (Twixes) [inline posthog/models/action/action.py:75]: It looks like at this point there are no more actions with `steps_json` being null in either US or EU. Can we make the field `steps_json` to enforce this for safety? This should be almost instant to execute

**Review by Twixes — APPROVED** [review]
Looks good

File: posthog/models/action/action.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/models/action/action.py:75]: Make sense. I'll do it as a one off follow up if thats alright

File: posthog/hogql/database/test/__snapshots__/test_database.ambr [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # serializer version: 1
 # name: TestDatabase.test_database_warehouse_joins_persons_poe_v2
-  'SELECT events__some_field.key AS key FROM events LEFT JOIN (SELECT groups.key AS key, key AS events__some_field___key FROM (SELECT groups.group_type_index AS index, groups.group_key AS key FROM groups WHERE equals(groups.team_id, 226) GROUP BY groups.group_type_index, groups.group_key) AS groups) AS events__some_field ON equals(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(events.person_properties, %(hogql_val_0)s), \'\'), \'null\'), \'^"|"$\', \'\'), events__some_field.events__some_field___key) WHERE equals(events.team_id, 226) LIMIT 10000'
+  'SELECT events__some_field.key AS key FROM events LEFT JOIN (SELECT groups.key AS key, key AS events__some_field___key FROM (SELECT groups.group_type_index AS index, groups.group_key AS key FROM groups WHERE equals(groups.team_id, 225) GROUP BY groups.group_type_index, groups.group_key) AS groups) AS events__some_field ON equals(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(events.person_properties, %(hogql_val_0)s), \'\'), \'null\'), \'^"|"$\', \'\'), events__some_field.events__some_field___key) WHERE equals(events.team_id, 225) LIMIT 10000'
```
Comment (pauldambra) [inline posthog/hogql/database/test/__snapshots__/test_database.ambr:3]: this team id should be being replaced or this test will flap 🙈 

(looks totally separate to this though)

**Review by pauldambra — APPROVED** [review]
ah @Twixes beat me to it 🤣 
well, have an extra tick

File: posthog/hogql/database/test/__snapshots__/test_database.ambr (hunk 2 above)
Comment (neilkakkar) [reply posthog/hogql/database/test/__snapshots__/test_database.ambr:3]: getting fixed.. soon

**Comment by Twixes** [comment]
Double the ticks, double the lyme disease 🎉

**Pull request merged**